twine/settings.py

```diff
@@ -121,6 +121,8 @@ def __init__(
         self.config_file = config_file
         self.comment = comment
         self.verbose = verbose
+        # Setting up logging before the config is loaded
+        utils.setup_logging(verbose)
```

Comment:
I'm not sure `utils` is the right place for `setup_logging`. If `--verbose` were a global option (similar to `--no-color`), I might suggest moving it `__init__.py` ala [flake8](https://github.com/PyCQA/flake8/blob/a7be77f761a4c29121d6bb6f61c11902281f9105/src/flake8/__init__.py#L27-L39), or a dedicated `twine.logging` module.

However, since `--verbose` only applies to commands that use `Settings` (i.e. `upload` and `register`, but not `check`), I'd tempted to move the logging configuration to this class, so that this becomes:

```suggestion
        self._setup_logging()
```

Reply:
I really don't know that it matters if `setup_logging` is in utils. I could see users of the the porcelain API  wanting to also turn off logging regardless of what's passed into the config and using this to try to do so (and thus wanting to use twine.utils)

Reply:
However, given our [suggestion](https://github.com/pypa/twine/pull/669/files#r446532787) to define a `twine.utils.logger` module attribute, I think that might be confusing with the current implementation, which also defines `logger` as a local variable in `setup_logging`. It looks like flake8 manages this by setting up logging in the package's [`__init__.py`](https://github.com/PyCQA/flake8/blob/master/src/flake8/__init__.py). @sigmavirus24 what do you think about following that pattern?

However, this might be confusing, once `logger` is defined as a module attribute. It looks like flake8 manages this by setting up logging in the package's [`__init__.py`](https://github.com/PyCQA/flake8/blob/master/src/flake8/__init__.py). @sigmavirus24 what do you think about following that pattern? For example:

```python
# twine/__init__.py
import logging

logger = logging.getLogger(__name__)

def setup_logging(verbose):
    # ...
    logger.setLevel(log_level)


# twine/settings.py
import twine

class Settings:
    def __init__(self, ...):
        # ...
        self.verbose = verbose
        twine.setup_logging(verbose)
```

Reply:
@bhrutledge I think your example will result in a circular dependency. I think doing 

```py
root_logger = logging.getLogger("twine")
```

is acceptable for altering the logger. I don't think we need to rely too heavily on `setup_logging` if we move that into `twine`. I'm also worried that `setup_logging` being at the top-level module will be confusing for folks looking at the API who aren't familiar with logging.

Reply:
Out of curiosity, I tried out the `twine/__init__.py` approach (plus some of the other suggested changes) in https://github.com/pypa/twine/commit/df46b3565ba24619c3b3fb5e4c8acbfbcdce3763, and it doesn't seem to result in a circular dependency.

Given that it's package-level configuration, `__init__.py` feels reasonable, but I see how it might not be obvious. That said, I've recently become wary of the `utils.py` pattern of unrelated behaviors. Maybe `cli.py` would be a more natural location? Or, how about a function in `settings.py`?

```python
import logging


def _setup_logging(verbose):
    # ...
    root_logger = logging.getLogger("twine")


class Settings:
    def __init__(self, ...):
        # ...
        self.verbose = verbose
        _setup_logging(verbose)
```

Here, I made it private, to avoid prematurely adding another public API.